File: src/entities/MovieHistory.ts
import { Column, Entity, JoinColumn, ManyToOne } from "typeorm";
import { Movie } from "./Movie";
import { Users } from "./Users";

@Entity("movie_history", { schema: "dionysus" })
export class MovieHistory {
  @Column("date", {
    name: "watch_date",
    nullable: true,
    default: () => "CURRENT_DATE",
  })
  watchDate: string | null;

  @Column("time without time zone", {
    name: "watch_time",
    nullable: true,
    default: () => "CURRENT_TIME",
  })
  watchTime: string | null;

  @Column("integer", { name: "minutes_watched", nullable: true })
  minutesWatched: number | null;

  @ManyToOne(() => Movie, (movie) => movie.movieHistories, {
    onDelete: "CASCADE",
    onUpdate: "CASCADE",
  })
  @JoinColumn([{ name: "movie_id", referencedColumnName: "movieId" }])
  movie: Movie;

  @ManyToOne(() => Users, (users) => users.movieHistories, {
    onDelete: "CASCADE",
    onUpdate: "CASCADE",
  })
  @JoinColumn([{ name: "user_id", referencedColumnName: "userId" }])
  user: Users;
}

File: engine/engine.api/Models/MovieHistory.cs
using System;
using System.Collections.Generic;

namespace engine.api.Models
{
    public partial class MovieHistory
    {
        public long UserId { get; set; }
        public long MovieId { get; set; }
        public DateOnly? WatchDate { get; set; }
        public TimeOnly? WatchTime { get; set; }
        public int? MinutesWatched { get; set; }

        public virtual Movie Movie { get; set; } = null!;
        public virtual User User { get; set; } = null!;
    }
}
